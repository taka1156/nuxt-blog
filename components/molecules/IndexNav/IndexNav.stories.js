import IndexNav from './IndexNav.vue';

export default {
  title: 'Molecules/IndexNav',
  component: IndexNav
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { IndexNav },
  template: '<index-nav v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { isOpen: false };
