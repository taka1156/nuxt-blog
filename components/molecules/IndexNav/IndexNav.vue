<template>
  <div>
    <div class="index-nav" @click="changeState">
      <base-nav-icon :is-open="isOpen">
        {{ isOpen ? 'CLOSE' : 'INDEX' }}
      </base-nav-icon>
    </div>
  </div>
</template>

<script>
import NavIcon from '../../atoms/BaseNavIcon/BaseNavIcon';
export default {
  name: 'IndexNav',
  components: {
    'base-nav-icon': NavIcon
  },
  props: {
    isOpen: {
      type: Boolean,
      default: false,
      required: true
    }
  },
  methods: {
    changeState() {
      this.$emit('change-state', !this.isOpen);
    }
  }
};
</script>

<style scoped>
.index-nav {
  position: fixed;
  right: 10px;
  bottom: 50px;
  z-index: 2;
  width: 50px;
  height: 55px;
  background-color: rgba(40, 167, 69, 0.9);
}

.index-nav__text {
  padding: 0;
  margin-top: -10px;
  font-size: 10px;
  color: white;
}
</style>
