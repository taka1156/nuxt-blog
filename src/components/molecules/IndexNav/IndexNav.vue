<template>
  <div>
    <div class="index-nav">
      <base-nav-icon :is-open="isOpen" @btn-click="changeState">
        {{ isOpen ? 'CLOSE' : 'INDEX' }}
      </base-nav-icon>
    </div>
  </div>
</template>

<script>
import NavIcon from '../../atoms/BaseNavIcon/BaseNavIcon';

export default {
  name: 'IndexNav',
  components: {
    'base-nav-icon': NavIcon
  },
  props: {
    /**
     * 目次の開閉状態
     */
    isOpen: {
      type: Boolean,
      required: true
    }
  },
  methods: {
    changeState() {
      this.$emit('change-state', !this.isOpen);
    }
  }
};
</script>

<style scoped>
::v-deep .base-nav-icon--extend {
  top: 0;
  margin: 0 auto;
}

.index-nav {
  position: fixed;
  right: 10px;
  bottom: 50px;
  z-index: 2;
  width: 50px;
  height: 55px;
  padding: 0;
  margin: auto;
  background-color: rgba(40, 167, 69, 0.9);
}
</style>
