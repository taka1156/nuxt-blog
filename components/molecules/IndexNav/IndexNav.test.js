import { mount } from '@vue/test-utils';
import IndexNav from './IndexNav.vue';

describe('IndexNav', () => {
  const indexNav = propsData =>
    mount(IndexNav, {
      propsData: {
        ...propsData
      }
    });

  it('IndexNav初期値: isOpen', () => {
    const wrapper = indexNav({ isOpen: false });
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
  });

  it('目次を開く', () => {
    const wrapper = indexNav({ isOpen: true });
    const navIconText = wrapper.find('.base-nav-icon__text');
    expect(wrapper.vm.isOpen).toBe(true);
    expect(navIconText.text()).toBe('CLOSE');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('目次を閉じる', () => {
    const wrapper = indexNav({ isOpen: false });
    const navIconText = wrapper.find('.base-nav-icon__text');
    expect(wrapper.vm.isOpen).toBe(false);
    expect(navIconText.text()).toBe('INDEX');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    const wrapper = indexNav({ isOpen: true });
    wrapper.vm.changeState();
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});
