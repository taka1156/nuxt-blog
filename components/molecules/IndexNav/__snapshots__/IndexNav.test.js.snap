// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexNav 目次を閉じる 1`] = `
"<div>
  <div class=\\"index-nav\\">
    <div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--close\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--close\\"></span> <span class=\\"base-nav-icon__text\\">
      INDEX
    </span></button></div>
  </div>
</div>"
`;

exports[`IndexNav 目次を開く 1`] = `
"<div>
  <div class=\\"index-nav\\">
    <div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--open\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle base-nav-icon__middle--fade\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--open\\"></span> <span class=\\"base-nav-icon__text\\">
      CLOSE
    </span></button></div>
  </div>
</div>"
`;
